extensions/amp-story/0.1/amp-story.css

```diff
@@ -312,7 +312,7 @@ amp-story-grid-layer * {
 }
 
 .i-amphtml-story-grid-template-thirds {
-  grid-template-rows: repeat(3, 1fr) !important;
+  grid-template-rows: 33% 33% 33% !important; /* `fr` is broken in Safari. */
```

Comment:
Not sure if it matters, but you might be able to use `calc(100% / 3)` to reduce the error of approximating thirds.